### PR TITLE
[IMP] various: improve user error wording

### PR DESCRIPTION
**Pull request by kcv-odoo**
Before this commit error messages in odoo are boring and not much attractive to user. Those were like odoo is preventing them from doing something user want to do.

In this commit, we have modified the message to be a more friendly and humorous tone, making it less tedious and more enjoyable. It aims to enhance the user experience and ensure that interactions with any application are both pleasant and informative. As a result, the messages have been clear, short, easy to understand and informative.

task-3422664

### CODE REVIEW

**Comment by rdeodoo** [comment]
Popcorn

**Review by JulienAlardot — CHANGES_REQUESTED** [review]
(expense only, not all accounting) Hate it as a concept, but let's do it right if we have to

File: addons/hr_holidays/models/hr_leave.py [hunk 1]
```diff
@@ -959,7 +959,7 @@ def write(self, values):
 
     @api.ondelete(at_uninstall=False)
     def _unlink_if_correct_states(self):
-        error_message = _('You cannot delete a time off which is in %s state')
+        error_message = _("You can't delete an %s time off")
```
Comment (h4818) [inline addons/hr_holidays/models/hr_leave.py:962]: You can't delete an Second Approval time off

`an` is used only when the following word starts with a vowel sound

File: addons/website/static/src/components/dialog/page_properties.xml [hunk 2]
```diff
@@ -61,20 +61,20 @@
 
 <t t-name="website.DeletePageDialog">
     <WebsiteDialog title="title" close="props.close">
-        <p t-if="props.resIds.length > 1">Are you sure you want to delete those pages?</p>
-        <p t-else="">Are you sure you want to delete this page?</p>
-        <p class="text-warning">Don't forget to update all links referring to it.</p>
+        <p t-if="props.resIds.length > 1">Ready to make those web pages disappear into thin air? Poof it will be gone forever!</p>
+        <p t-else="">Ready to make this web page disappear into thin air? Poof it will be gone forever!</p>
+        <p class="text-warning">But before deleting it, make sure you update all links referring to it. That way, your customers won't bump into the not-so-famous 404 error page..</p>
```
Comment (rdeodoo) [inline addons/website/static/src/components/dialog/page_properties.xml:66]: Weren't those messages supposed to be only for "rare" error cases that you should not encounter with basic flow?
Here, you will face this message pretty much everytime you delete a page IMO, pages URL are often cross referenced, or in the website menu.
I think the change in line 66 was enough to bring a touch of "fun" while not being too invasive. But I am fine to keep the whole change if you prefer, it's fine.

**Review by rdeodoo — APPROVED** [review]
Approving the very small changes in website, just a question out of curiosity but you can keep it as it is.

File: addons/sale/models/sale_order.py [hunk 3]
```diff
@@ -1183,13 +1183,9 @@ def _get_invoice_grouping_keys(self):
...
         return _(
-            "Cannot create an invoice. No items are available to invoice.\n\n"
-            "To resolve this issue, please ensure that:\n"
-            "   \u2022 The products have been delivered before attempting to invoice them.\n"
-            "   \u2022 The invoicing policy of the product is configured correctly.\n\n"
-            "If you want to invoice based on ordered quantities instead:\n"
-            "   \u2022 For consumable or storable products, open the product, go to the 'General Information' tab and change the 'Invoicing Policy' from 'Delivered Quantities' to 'Ordered Quantities'.\n"
-            "   \u2022 For services (and other products), change the 'Invoicing Policy' to 'Prepaid/Fixed Price'.\n"
+            "There’s nothing to invoice!\n\n"
+            "Tip: Quantities to invoice appear in blue in the invoice form. If you have none, you might consider delivering your customer before invoicing :)\n"
```
Comment (Feyensv) [inline addons/sale/models/sale_order.py:1187]: Quantities to invoice appear in blue on the invoice form ? :thinking: 
We're on the SO here, not on the invoice, this message is misleading.

It was already changed one month ago to make it clearer and more understandable ... (3b7dd79e5f44f76380920e935c7d079ad0f056ed)
Let's discard the translator's work once again :partying_face: 

cc: @dalaOdoo @xavierbol

File: addons/phone_validation/tools/phone_validation.py [hunk 4]
```diff
@@ -22,7 +22,7 @@ def phone_parse(number, country_code):
             ) from e
 
         if not phonenumbers.is_possible_number(phone_nbr):
-            raise UserError(_('Impossible number %s: probably invalid number of digits.', number))
+            raise UserError(_("The number %s has an invalid number of digits, unless you're dialing aliens. Let's fix it for this dimension!", number))
```
Comment (MissingNoShiny) [inline addons/phone_validation/tools/phone_validation.py:25]: I don't see the relationship between aliens and dimensions (didn't think I would ever have to comment something like this in a review)

File: addons/stock/models/stock_move_line.py [hunk 5]
```diff
@@ -475,7 +475,11 @@ def write(self, vals):
     def _unlink_except_done_or_cancel(self):
         for ml in self:
             if ml.state in ('done', 'cancel'):
-                raise UserError(_('You can not delete product moves if the picking is done. You can only correct the done quantities.'))
+                raise UserError(_(
+                    "Deleting product moves after a picking is done?\n\n"
+                    "That would be like going back in time to revert all operations triggered after this move. You better not do it, Marty!\n\n"
```
Comment (MissingNoShiny) [inline addons/stock/models/stock_move_line.py:480]: A lot of customers will not get this reference

File: addons/web/static/src/views/list/list_confirmation_dialog.xml [hunk 6]
```diff
@@ -10,7 +10,8 @@
                         Among the <t t-esc="props.nbRecords"/> selected records,
                         <t t-esc="props.nbValidRecords"/> are valid for this update.<br/>
                     </t>
-                    Are you sure you want to perform the following update on those <t t-esc="props.nbValidRecords"/> records?
+                    Are you sure you want to update <b><t t-esc="props.nbValidRecords"/> records</b>?
+                    <div class="text-muted"><i>Enjoy how well designed Odoo is :)</i></div>
```
Comment (MissingNoShiny) [inline addons/web/static/src/views/list/list_confirmation_dialog.xml:14]: ?

File: addons/website_slides_survey/models/survey_survey.py [hunk 7]
```diff
@@ -34,7 +34,8 @@ def _unlink_except_linked_to_course(self):
         if certifications:
             certifications_course_mapping = [_('- %s (Courses - %s)', certi.title, '; '.join(certi.slide_channel_ids.mapped('name'))) for certi in certifications]
             raise ValidationError(_(
-                'Any Survey listed below is currently used as a Course Certification and cannot be deleted:\n%s',
+                'Uh-oh! You can’t delete a survey used as a Course Certification! Otherwise students might think diplomas just grow on trees.\n'
+                'The courses that need it:\n%s',
```
Comment (MissingNoShiny) [inline addons/website_slides_survey/models/survey_survey.py:38]:
```suggestion
                'The courses that need it are:\n%s',
```

File: addons/hr_timesheet/models/project_task.py [hunk 8]
```diff
@@ -234,9 +234,9 @@ def _unlink_except_contains_entries(self):
         task_with_timesheets_ids = [task.id for task, in timesheet_data]
         if task_with_timesheets_ids:
             if len(task_with_timesheets_ids) > 1:
-                warning_msg = _("These tasks have some timesheet entries referencing them. Before removing these tasks, you have to remove these timesheet entries.")
+                warning_msg = _("Some timesheet entries are hitched to these tasks! Remove them first, then you’ll be able to delete it!")
```
[the remark this answers is not in the document]
Comment (MissingNoShiny) [reply addons/hr_timesheet/models/project_task.py:237]: Also note that we don't handle singular/plural in messages like this, it might be a good time to get rid of the other one to only keep this one

**Review by MissingNoShiny — COMMENTED** [review]
Thanks for your work, I left a few comments/suggestions :slightly_smiling_face:

File: addons/account/models/account_move.py [hunk 9]
```diff
@@ -1838,10 +1838,11 @@ def _check_balanced(self, container):
...
                 error_msg += _(
-                    "\n\n"
-                    "The move (%s) is not balanced.\n"
-                    "The total of debits equals %s and the total of credits equals %s.\n"
-                    "You might want to specify a default account on journal \"%s\" to automatically balance each move.",
+                    '\n\n'
+                    'Unbalanced move alert! The move %s is feeling a bit unbalanced, with %s on '
+                    'the debit side and %s on the credit side. We need some equilibrium here!\n\n'
+                    'Consider adding a default account on the journal \"%s\" to '
+                    'automatically balance each move and restore order to the accounting universe!',
```
Comment (oomsveta) [inline addons/account/models/account_move.py:1845]: Actual improvement: use keyworded placeholders. In other words, you should use `%(move)s`, `%(total_debit)s`, `%(total_credit)s`... instead of plain `%s`.

Translating text in a language sometimes involves reordering words. Positional arguments are always inserted in the same order, which order matches English syntax, but this isn't necessarily right for every syntax of every language in the world. Whenever there is more than one “format specifier” (%s), you should always use keyworded arguments so that the translators can freely reorder the words without any problems.

Not to mention that naming your values also provides more context, which is a truly valuable hint for the translators to translate your terms correctly 👍️

File: addons/account/models/account_move.py [hunk 10]
```diff
@@ -3682,7 +3683,10 @@ def _post(self, soft=True):
 
             if not invoice.partner_id:
                 if invoice.is_sale_document():
-                    raise UserError(_("The field 'Customer' is required, please complete it to validate the Customer Invoice."))
+                    raise UserError(_(
+                        "The 'Customer' field is required to validate the invoice.\n"
+                        "You probably don’t want to explain to your auditor that you invoiced the invisible man :)"
```
Comment (oomsveta) [inline addons/account/models/account_move.py:3688]: any reason to use `’` rather than `'`?

File: addons/account/models/account_move.py [hunk 11]
```diff
@@ -3698,7 +3702,10 @@ def _post(self, soft=True):
             if move.state in ['posted', 'cancel']:
                 raise UserError(_('The entry %s (id %s) must be in draft.', move.name, move.id))
             if not move.line_ids.filtered(lambda line: line.display_type not in ('line_section', 'line_note')):
-                raise UserError(_('You need to add a line before posting.'))
+                raise UserError(_(
+                    "Make sure you add a line before posting the invoice. Invoices can't be created from thin air.\n"
+                    "You’ll thank me later for avoiding a weird explanation to your auditor :)"
```
Comment (oomsveta) [inline addons/account/models/account_move.py:3707]: Mixed apostrophes: `'` and `’`. I'd advocate picking either one and sticking with it

File: addons/account/models/account_payment_term.py [hunk 12]
```diff
@@ -222,7 +222,7 @@ def _compute_terms(self, date_ref, currency, company, tax_amount, tax_amount_cur
     @api.ondelete(at_uninstall=False)
     def _unlink_except_referenced_terms(self):
         if self.env['account.move'].search([('invoice_payment_term_id', 'in', self.ids)]):
-            raise UserError(_('You can not delete payment terms as other records still reference it. However, you can archive it.'))
+            raise UserError(_('Uh-oh! Those payment terms are quite popular and can’t be deleted, as there are still some records referencing it. How about archiving it instead?'))
```
Comment (oomsveta) [inline addons/account/models/account_payment_term.py:225]:
```suggestion
            raise UserError(_('Uh-oh! Those payment terms are quite popular and can’t be deleted, as there are still some records referencing them. How about archiving them instead?'))
```

File: addons/account/wizard/account_payment_register.py [hunk 13]
```diff
@@ -652,7 +652,7 @@ def default_get(self, fields_list):
 
             # Check.
             if not available_lines:
-                raise UserError(_("You can't register a payment because there is nothing left to pay on the selected journal items."))
+                raise UserError(_("There's nothing left to pay on those selected journal items, so no payment registration necessary. You've got your finances under control like a boss!"))
```
Comment (oomsveta) [inline addons/account/wizard/account_payment_register.py:655]: missing word?
```suggestion
                raise UserError(_("There's nothing left to pay on those selected journal items, so no payment registration is necessary. You've got your finances under control like a boss!"))
```

File: addons/hr_timesheet/models/project_task.py (hunk 8 above)
Comment (oomsveta) [inline addons/hr_timesheet/models/project_task.py:237]:
```suggestion
                warning_msg = _("Some timesheet entries are hitched to these tasks! Remove them first, then you’ll be able to delete them!")
```

File: addons/hr_timesheet/models/project_task.py [hunk 14]
```diff
@@ -234,9 +234,9 @@ def _unlink_except_contains_entries(self):
...
             else:
-                warning_msg = _("This task has some timesheet entries referencing it. Before removing this task, you have to remove these timesheet entries.")
+                warning_msg = _("Some timesheet entries are hitched to this task! Remove them first, then you’ll be able to delete it!")
```
Comment (oomsveta) [inline addons/hr_timesheet/models/project_task.py:239]:
```suggestion
                warning_msg = _("Some timesheet entries are hitched to this task! Remove them first, then you’ll be able to delete them!")
```

File: addons/mail/i18n/mail.pot [hunk 15]
```diff
@@ -10901,4 +10910,4 @@ msgstr ""
 #: code:addons/mail/models/discuss/discuss_channel_member.py:0
 #, python-format
 msgid "“%(member_name)s” in “%(channel_name)s”"
-msgstr ""
+msgstr ""
```
Comment (oomsveta) [inline addons/mail/i18n/mail.pot:10913]: Missing newline at end of file.

Newlines are important; according to the UNIX Standard, a line is "a sequence of zero or more non-\<newline> characters plus a terminating \<newline> character". This means that without the newline at the end of the file, the last "line" is not considered a line in the UNIX sense. This can sometimes be a problem with some more pedantic programs: `wc`, for example, wouldn't count your last line.

File: addons/repair/i18n/repair.pot [hunk 16]
```diff
@@ -358,7 +360,7 @@ msgstr ""
 
 #. module: repair
 #: model_terms:ir.ui.view,arch_db:repair.stock_warn_insufficient_qty_repair_form_view
-msgid "Do you confirm you want to repair"
+msgid "<br/>You can still confirm the repair of"
```
Comment (oomsveta) [inline addons/repair/i18n/repair.pot:363]: Looks like you got your text split in two; that makes it hard to translate. Not that it was any better before your changes, though.

File: addons/stock/models/stock_picking.py [hunk 17]
```diff
@@ -1192,8 +1192,8 @@ def _get_without_quantities_error_message(self):
         :rtype: str
         """
         return _(
-            'You cannot validate a transfer if no quantities are reserved. '
-            'To force the transfer, encode quantities.'
+            "Transfer trouble alert! 0 quantity transfer? That's like trying to move air around!\n"
```
Comment (oomsveta) [inline addons/stock/models/stock_picking.py:1195]:
```suggestion
            "Transfer trouble alert! Zero quantity transfer? That's like trying to move air around!\n"
```

File: addons/website/static/src/components/dialog/page_properties.xml [hunk 18]
```diff
@@ -61,20 +61,20 @@
 
 <t t-name="website.DeletePageDialog">
     <WebsiteDialog title="title" close="props.close">
-        <p t-if="props.resIds.length > 1">Are you sure you want to delete those pages?</p>
-        <p t-else="">Are you sure you want to delete this page?</p>
-        <p class="text-warning">Don't forget to update all links referring to it.</p>
+        <p t-if="props.resIds.length > 1">Ready to make those web pages disappear into thin air? Poof it will be gone forever!</p>
+        <p t-else="">Ready to make this web page disappear into thin air? Poof it will be gone forever!</p>
```
Comment (oomsveta) [inline addons/website/static/src/components/dialog/page_properties.xml:65]: Don't use "poof," as it is also a British slur. Of course, no human would interpret that as the cursed word, right? The thing is, some people machine-translate Odoo for their own use. We already got a ticket from a disenchanted customer after he was greeted with a very bad homophobic slur in Odoo 17. We resolved to remove it at once. Please don't bring back this disastrous wording

**Review by oomsveta — COMMENTED** [review]
Hello, thank you for your work! This seems to be a good improvement overall 🙂

File: addons/crm/models/crm_lead.py [hunk 19]
```diff
@@ -1398,7 +1398,9 @@ def _merge_opportunity(self, user_id=False, team_id=False, auto_unlink=True, max
             raise UserError(_('Please select more than one element (lead or opportunity) from the list view.'))
 
         if max_length and len(self.ids) > max_length and not self.env.is_superuser():
-            raise UserError(_("To prevent data loss, Leads and Opportunities can only be merged by groups of %(max_length)s.", max_length=max_length))
+            raise UserError(_(
+                "Data safety alert! To prevent any disappearing act, Leads and Opportunities can only be "
+                "merged in groups of %(max_length)s. So gather your fabulous fivesome!", max_length=max_length))
```
[the remark this answers is not in the document]
Comment (tde-banana-odoo) [reply addons/crm/models/crm_lead.py:1403]: This doesn't make sense at all actually.

File: addons/phone_validation/tools/phone_validation.py (hunk 4 above)
Comment (tde-banana-odoo) [reply addons/phone_validation/tools/phone_validation.py:25]: Please make an understandable error, moreover while changing this remove the 'invalid number of digits' part, like
```suggestion
            raise UserError(_("The number %s does not appear to be a valid number. [insert bad joke here]", number))
```

File: addons/website_slides_survey/models/survey_survey.py (hunk 7 above)
Comment (tde-banana-odoo) [reply addons/website_slides_survey/models/survey_survey.py:38]: Please remove the tree part, makes no sense. You don't HAVE to find a bad joke, especially this flow is accessible only to survey managers.

**Review by tde-banana-odoo — CHANGES_REQUESTED** [review]
SM apps review

File: addons/hr_holidays/models/hr_leave.py (hunk 1 above)
Comment (sofiagvaladze) [reply addons/hr_holidays/models/hr_leave.py:962]: New version sounds weird.. 

I prefer old better.. considering the state names

File: addons/hr_holidays/models/hr_leave.py [hunk 20]
```diff
@@ -972,7 +972,7 @@ def _unlink_if_correct_states(self):
                 if hol.sudo().employee_ids and not hol.employee_id:
                     raise UserError(_('You cannot delete a time off assigned to several employees'))
         else:
-            for holiday in self.filtered(lambda holiday: holiday.state not in ['draft', 'cancel', 'confirm']):
+            for holiday in self.filtered(lambda holiday: holiday.state not in ['draft', 'cancel', 'confirm', 'refuse']):
```
Comment (sofiagvaladze) [inline addons/hr_holidays/models/hr_leave.py:975]: why this change? it's a bit beyond wording changes

File: addons/hr_holidays/models/hr_leave.py [hunk 21]
```diff
@@ -1137,7 +1137,7 @@ def action_approve(self, check_state=True):
 
         # Do not check the state in case we are redirected from the dashboard
         if check_state and any(holiday.state != 'confirm' for holiday in self):
-            raise UserError(_('Time off request must be confirmed ("To Approve") in order to approve it.'))
+            raise UserError(_('You can only manage time off requests that are in state “To approve”'))
```
Comment (sofiagvaladze) [inline addons/hr_holidays/models/hr_leave.py:1140]: manage is a bit more general word. It's confusing.. we are confirming the time off, so let's call it so

File: addons/stock/models/stock_picking.py [hunk 22]
```diff
@@ -1192,8 +1192,8 @@ def _get_without_quantities_error_message(self):
...
+            "Set the quantities, and let the magic happen!"
```
Comment (ticodoo) [inline addons/stock/models/stock_picking.py:1196]: I think this sounds better. Also, I'm not a big fan of the "trying to move air around" joke since some business may literally be selling/moving (tanks of) air, but I can't think of a better one to replace it with.

```suggestion
            "Trouble alert! Transferring no quantities? That's like trying to move air around!\n"
            "Set some quantities, and let the magic happen!"
```

File: addons/stock/wizard/stock_warn_insufficient_qty_views.xml [hunk 23]
```diff
@@ -6,7 +6,7 @@
         <field name="arch" type="xml">
             <form>
                 <div>
-                    The product is not available in sufficient quantity
+                    The product is not available in sufficient quantity.
```
Comment (ticodoo) [inline addons/stock/wizard/stock_warn_insufficient_qty_views.xml:9]: ? There more to the sentence after "quantity" when multi-locations are active, so there is purposely no .
image

**Review by ticodoo — COMMENTED** [review]
I only reviewed the logistics related changes. There are probably more English issues with the others :shrug:

**Review by pkyriakou — COMMENTED** [review]
Alright thank you @kpt-odoo and @kcv-odoo for your work :100: 
I left some comments to be looked into. They are both related to the content of the messages but also some grammatical / syntactical things. 
I understand that the former type of comments is subjective so I have indicated them as [content feedback]. You don't have to perform any changes unless Jihane also approves them. 
The latter type of comments are more objective and thus can be addressed as is but if you feel that I'm reaching, you can still ask about Jihane's opinion. 
That's all :v:

File: addons/repair/wizard/stock_warn_insufficient_qty_views.xml [hunk 24]
```diff
@@ -6,7 +6,7 @@
         <field name="inherit_id" ref="stock.stock_warn_insufficient_qty_form_view"/>
         <field name="arch" type="xml">
             <xpath expr="//div[@name='description']" position="after">
-                Do you confirm you want to repair <strong><field name="quantity" readonly="True"/></strong><field name="product_uom_name" readonly="True" class="mx-1"/>from location <strong><field name="location_id" readonly="True"/></strong>? This may lead to inconsistencies in your inventory.
+                <br/>You can still confirm the repair of <strong><field name="quantity" readonly="True"/></strong><field name="product_uom_name" readonly="True" class="mx-1"/>from <strong><field name="location_id" readonly="True"/></strong>, but get ready to have an adventure taking inventory to fix the negative stock quantity!
```
[the remark this answers is not in the document]
Comment (ticodoo) [reply addons/repair/wizard/stock_warn_insufficient_qty_views.xml:9]: It's a tricky one indeed. Inventory = both a noun and a verb depending on how it's used. I already commented on how this change is misleading though for non-admin users (they can't confirm inventory counts without an admin) and I agree that the "adventure" part makes the sentence harder to read. I couldn't think of a better way to word the sentence without dropping it (and making it sound less friendly or equally hard to read):

You can still confirm the repair of _____ from _______, but get ready to have to take inventory to fix the negative stock quantity!

The best I can suggest is rewording everything after the ", but"

**Review by pkyriakou — COMMENTED** [review]
Thank you very much for your work :100: 
I did leave one comment which was an oversight in my previous suggestion but other than that I couldn't spot any grammatical errors. I will green the task regardless.

**Review by Bertrand2 — APPROVED** [review]

**Comment by rdeodoo** [comment]
Whatever happen to this one, don't forget to retarget master, we don't want to break translations that late after the release.

**Comment by ppr-odoo** [comment]
Continue at: https://github.com/odoo/odoo/pull/178821

**Comment by aab-odoo** [comment]
Closing as re-targeted to https://github.com/odoo/odoo/pull/178821
